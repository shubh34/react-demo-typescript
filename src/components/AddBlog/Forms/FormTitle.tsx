import React from 'react';
import { Form, Input } from 'semantic-ui-react';

interface IFormTitle {
	title: string;
	onChange: Function;
	
}
const FormTitle: React.FC<IFormTitle> = ({ title, onChange }) => {
	const onTitleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
		onChange(e.target.name, e.target.value);
	};
	return (
		<Form.Field>
			<label>Title</label>
			<Input
				spellCheck
				type='text'
				name='title'
				placeholder='Enter your blog title'
				onChange={onTitleChange}
				value={title}
			/>
		</Form.Field>
	);
};

export default FormTitle;
