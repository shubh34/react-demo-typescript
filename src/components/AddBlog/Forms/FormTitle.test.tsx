import React from 'react';
import { shallow } from 'enzyme';
import FormTitle from './FormTitle';


describe('FormTitle', () => {
    it('renders without crashing', () => {
        const wrapper = shallow(<FormTitle title="My Blog Tiltle" onChange={jest.fn()} />);
        expect(wrapper).toMatchSnapshot();
    });

    it('should call onChange', () => {
        const onChange = jest.fn()
        const wrapper = shallow(<FormTitle title="My Blog Tiltle" onChange={onChange} />);
        wrapper.find('Input').simulate('change', {
            target: {
                name: 'Title',
                value: 'test'
            }});
        expect(onChange).toBeCalledWith('Title', 'test')
    });
})

