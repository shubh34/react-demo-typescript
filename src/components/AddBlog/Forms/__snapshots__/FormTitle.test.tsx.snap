// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormTitle renders without crashing 1`] = `
<FormField>
  <label>
    Title
  </label>
  <Input
    name="title"
    onChange={[Function]}
    placeholder="Enter your blog title"
    spellCheck={true}
    type="text"
    value="My Blog Tiltle"
  />
</FormField>
`;
